src/ansys/mapdl/core/cli/login.py

```diff
@@ -0,0 +1,215 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Login into a PyHPS cluster"""
+from getpass import getpass
+import logging
+from typing import Optional
+
+import click
+
+logger = logging.getLogger()
+
+# logging.basicConfig(
+#     level=logging.DEBUG,
+#     format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+#     # handlers=[
+#     #     logging.FileHandler("pymapdl.log"),
+#     #     logging.StreamHandler()
+#     # ]
+# )
+
+
+@click.command(
+    name="login",
+    short_help="Login into an HPS cluster.",
+    help="""Login into an HPS cluster.
+
+It does store credentials (cluster url, password and username) in the OS credential manager.
+If you want to change any credential, just issue the command again with the new values.
+
+""",
+)
+@click.option("--user", default=None, type=str, help="The username to login.")
+@click.option("--password", default=None, type=str, help="The password to login.")
+@click.option(
+    "--url",
+    default=None,
+    type=str,
+    help="The HPS cluster URL. For instance 'https://10.231.106.1:3000/hps'.",
+)
+@click.option(
+    "--default",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Set the default user, password and URL. These credentials are not tested.""",
+)
+@click.option(
+    "--test_token",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Test if the token is valid. This argument is ignored if '--default' argument is ``True``.""",
+)
+@click.option(
+    "--quiet",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Suppress al console printout.""",
+)
+def login(
+    user: Optional[str] = None,
+    password: Optional[str] = None,
+    url: Optional[str] = None,
+    default: bool = False,
+    test_token: bool = False,
+    quiet: bool = False,
+):
+    """
+    Command Line Interface for logging in and getting an access token.
+
+    Parameters
+    user (str): Username for login
+    password (str): Password for login
+    default (bool): If used, the input data is set as default and it will be used for any login which does not specify the url, user or password.
+    """
+    from ansys.mapdl.core.hpc.login import login_in_cluster, store_credentials
+
+    if quiet:
+        import urllib3
+
+        urllib3.disable_warnings()
+
+    logger.debug("Storing non-default credentials.")
+    if not user:
```

Comment:
I would have assumed a "default" check would short circuit prompting for user/pass

Reply:
The CLI is aimed to work as `docker login`. It does store the user and password in the OS credential manager. It does not login, it just store them. 

Having a default is a way to have an user, and password to use when you have only the URL (useful for externally managed credential systems; multiple clusters, same IDs).